rkt/status.go

```diff
@@ -203,38 +203,24 @@ func printStatus(p *pkgPod.Pod) error {
 		stdout.Printf("networks=%s", fmtNets(p.Nets))
 	}
 
-	if state == pkgPod.Running || state == pkgPod.Deleting || state == pkgPod.ExitedDeleting || state == pkgPod.Exited || state == pkgPod.ExitedGarbage {
-		var pid int
-		pidCh := make(chan int, 1)
-
-		// Wait slightly because the pid file might not be written yet when the state changes to 'Running'.
-		go func() {
-			for {
-				pid, err := p.Pid()
-				if err == nil {
-					pidCh <- pid
-					return
-				}
-				time.Sleep(time.Millisecond * 100)
-			}
-		}()
-
-		select {
-		case pid = <-pidCh:
-		case <-time.After(time.Second):
-			return fmt.Errorf("unable to get PID for pod %q: %v", p.UUID, err)
-		}
+	if !(state == pkgPod.Running || state == pkgPod.Deleting || state == pkgPod.ExitedDeleting || state == pkgPod.Exited || state == pkgPod.ExitedGarbage) {
+		return nil
+	}
 
-		stdout.Printf("pid=%d\nexited=%t", pid, (state == pkgPod.Exited || state == pkgPod.ExitedGarbage))
+	if pid, err := p.Pid(); err == nil {
```

Comment:
I think this change may re-introduce the race documented in the comment below, which was previously addressed by the goroutine+timeout. I would suggest to special-case the running case, block on it until pid exists or we hit the timeout, and then proceed with your logic.